### PR TITLE
feat: Region-based tuner support for P5en

### PR DESCRIPTION
**Pull request by arunkarthik-akkart**
This patch has the below changes for the region based tuner for p5en
 	- Enable region based tuner for P5en.
	- add vertices for region based tuner for all_reduce bitmask 0x0 and 0x7.
	- add vertices for region based tuner for all_gather bitmask 0x7.
	- add vertices for region based tuner for reduce_scatter bitmask 0x7.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sunkuamzn** [comment]
bot:aws:retest

File: tests/unit/show_tuner_decisions.c [hunk 1]
```diff
@@ -30,7 +30,7 @@ int main(int argc, const char **argv)
 				return 1;
 			}
 
-			for (size_t nmibytes = 1; nmibytes <= 32 * 1024; nmibytes <<= 1) {
+			for (size_t nmibytes = 1; nmibytes <= 32 * 1024 * 1024; nmibytes <<= 1) {
```
Comment (bwbarrett) [inline tests/unit/show_tuner_decisions.c:33]: This variable should be named `nkibytes` now.

Comment (arunkarthik-akkart) [reply tests/unit/show_tuner_decisions.c]: Fixed this.

**Review by bwbarrett — APPROVED** [review]

**Comment by a-szegel** [comment]
bot:aws:retest ... infra issues, cluster disk space full

**Comment by a-szegel** [comment]
bot:aws:retest

**Review by xwangxin — APPROVED** [review]

**Pull request merged**